File: src/main/java/com/epri/dss/common/impl/PowerTerminal.java
package com.epri.dss.common.impl;

import com.epri.dss.common.Conductor;
import com.epri.dss.common.Terminal;

public class PowerTerminal implements Terminal {

	private int numCond;
	private int activeConductor;

	protected int busRef;

	protected int[] termNodeRef;
	protected Conductor[] conductors;
	protected boolean checked;

	public PowerTerminal(int nCond) {
		super();
		this.numCond = nCond;
		this.busRef = -1;  // signify not set
		this.termNodeRef = new int[this.numCond];
		this.conductors = new Conductor[this.numCond];
		for (int i = 0; i < this.numCond; i++)
			conductors[i] = new ConductorImpl();
		activeConductor = 0;  // TODO Check zero based indexing
	}

	public int getBusRef() {
		return busRef;
	}

	public void setBusRef(int value) {
		busRef = value;
	}

	public int[] getTermNodeRef() {
		return termNodeRef;
	}

	public void setTermNodeRef(int[] value) {
		termNodeRef = value;
	}

	public Conductor[] getConductors() {
		return conductors;
	}

	public void setConductors(Conductor[] value) {
		conductors = value;
	}

	public boolean isChecked() {
		return checked;
	}

	public void setChecked(boolean value) {
		checked = value;
	}

	public void setActiveConductor(int value) {
		if ((value >= 0) & (value < numCond))  // TODO Check zero based indexing
			activeConductor = value;
	}

	public int getActiveConductor() {
		return activeConductor;
	}
}

File: pydss/common/PowerTerminal.py
from dss.common.Terminal import (Terminal,)
from dss.common.impl.ConductorImpl import (ConductorImpl,)


class PowerTerminal(object, Terminal):
    numCond = None
    activeConductor = None
    busRef = None
    termNodeRef = None
    conductors = None
    checked = None

    def __init__(self, nCond):
        super(PowerTerminal, self)()
        self.numCond = nCond
        self.busRef = -1
        # signify not set
        self.termNodeRef = [None] * self.numCond
        self.conductors = [None] * self.numCond
        _0 = True
        i = 0
        while True:
            if _0 is True:
                _0 = False
            else:
                i += 1
            if not (i < self.numCond):
                break
            self.conductors[i] = ConductorImpl()
        self.activeConductor = 0
        # TODO Check zero based indexing

    def getBusRef(self):
        return self.busRef

    def setBusRef(self, value):
        self.busRef = value

    def getTermNodeRef(self):
        return self.termNodeRef

    def setTermNodeRef(self, value):
        self.termNodeRef = value

    def getConductors(self):
        return self.conductors

    def setConductors(self, value):
        self.conductors = value

    def isChecked(self):
        return self.checked

    def setChecked(self, value):
        self.checked = value

    def setActiveConductor(self, value):
        if (value >= 0) & (value < self.numCond):
            # TODO Check zero based indexing
            self.activeConductor = value

    def getActiveConductor(self):
        return self.activeConductor
